group-coordinator/src/main/java/org/apache/kafka/coordinator/group/RecordSerializer.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.coordinator.group;
+
+import org.apache.kafka.common.protocol.MessageUtil;
+import org.apache.kafka.coordinator.group.runtime.PartitionWriter;
+
+/**
+ * Serializer which serializes {{@link Record}} to bytes.
+ */
+public class RecordSerializer implements PartitionWriter.Serializer<Record> {
+    @Override
+    public byte[] serializeKey(Record record) {
+        // Record does not accept a null key.
+        return MessageUtil.toVersionPrefixedBytes(
```

Comment:
Are there other ways to serialize? Just curious if we had something like this already.

Reply:
We use the very same code `MessageUtil.toVersionPrefixedBytes` in scala.

Reply:
I guess I was wondering why we have a unique class here for this. It makes it seem like we would want other implementations.

Reply:
Yeah. I made it generic in the case we would want to reuse it for another coordinator in the future. This one is tight to the Record class.